src/main/java/org/gridsuite/voltageinit/server/service/NetworkModificationService.java

```diff
@@ -0,0 +1,116 @@
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpStatusCodeException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.util.Objects;
+import java.util.UUID;
+
+/**
+ * @author Franck Lecuyer <franck.lecuyer at rte-france.com>
+ */
+@Service
+public class NetworkModificationService {
+    private static final String NETWORK_MODIFICATION_API_VERSION = "v1";
+    private static final String DELIMITER = "/";
+    private static final String GROUP_PATH = "groups" + DELIMITER + "{groupUuid}";
+    public static final String QUERY_PARAM_ERROR_ON_GROUP_NOT_FOUND = "errorOnGroupNotFound";
+
+    private String networkModificationServerBaseUri;
+
+    private final RestTemplate restTemplate = new RestTemplate();
+
+    private final ObjectMapper objectMapper;
+
+    @Autowired
+    NetworkModificationService(@Value("${gridsuite.services.network-modification-server.base-uri:http://network-modification-server/}") String networkModificationServerBaseUri,
+                               ObjectMapper objectMapper) {
+        this.networkModificationServerBaseUri = networkModificationServerBaseUri;
+        this.objectMapper = objectMapper;
+    }
+
+    public void setNetworkModificationServerBaseUri(String networkModificationServerBaseUri) {
+        this.networkModificationServerBaseUri = networkModificationServerBaseUri + DELIMITER;
+    }
+
+    private String getNetworkModificationServerURI() {
+        return this.networkModificationServerBaseUri + DELIMITER + NETWORK_MODIFICATION_API_VERSION + DELIMITER;
+    }
+
+    public void deleteModificationsGroup(UUID groupUUid) {
+        Objects.requireNonNull(groupUUid);
+        var path = UriComponentsBuilder.fromPath(GROUP_PATH)
+            .queryParam(QUERY_PARAM_ERROR_ON_GROUP_NOT_FOUND, false)
+            .buildAndExpand(groupUUid)
+            .toUriString();
+
+        try {
+            restTemplate.delete(getNetworkModificationServerURI() + path);
+        } catch (HttpStatusCodeException e) {
+            throw new PowsyblException("Error deleting modifications group", e);
+        }
+    }
+
+    public UUID createVoltageInitModificationGroup(OpenReacResult result) {
+        UUID modificationsGroupUuid = null;
+
+        try {
+            VoltageInitModificationInfos voltageInitModificationInfos = new VoltageInitModificationInfos();
+
+            result.getGeneratorModifications().forEach(gm -> {
+                if (gm.getModifs().getTargetV() != null || gm.getModifs().getTargetQ() != null) {
+                    GeneratorModificationInfos.GeneratorModificationInfosBuilder builder = GeneratorModificationInfos.builder()
+                        .generatorId(gm.getGeneratorId());
+                    if (gm.getModifs().getTargetV() != null) {
+                        builder.voltageSetpoint(gm.getModifs().getTargetV());
```

Comment:
Could we use `@JsonInclude(Include.NON_NULL)` instead maybe ?

Reply:
Done